File: 0001_0100/0051/0051.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = f.read().split()
    n = int(n)
    k = len(k)
    g = n - k
    while g > 0:
        n *= g
        g -= k
    q.write(str(n))

        

File: 0001_0100/0051/0051.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s string
	var n int
	fmt.Fscan(f, &n, &s)
	k := len(s)
	for i := n - k; i > 0; i -= k {
		n *= i
	}
	fmt.Fprint(q, n)
}
